pilot/pkg/proxy/envoy/v2/discovery.go

```diff
@@ -381,6 +381,13 @@ func (s *DiscoveryServer) ConfigUpdate(full bool) {
 // It ensures that at minimum minQuiet time has elapsed since the last event before processing it.
 // It also ensures that at most maxDelay is elapsed between receiving an event and processing it.
 func (s *DiscoveryServer) handleUpdates(stopCh <-chan struct{}) {
+	debounce(s.updateChannel, stopCh, func(full bool) {
+		go s.doPush(full)
```

Comment:
why not directly pass `s.doPush`?

Reply:
Because we want this to be `go s.doPush(full)` not `s.doPush(full)`

Reply:
this can be done in debounce func

Reply:
the tests don't trigger a goroutine for pushfn